Integrate Todo List Into Dashboard

sbodmer01 opened the pull request:
**Please check if the PR fulfills these requirements:**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
- Bug Fix/New Feature: Previously, the integrated todo list would continue growing as tasks were updated, growing even outside of the site card. To fix this, I added a scrolling function so that the todo list has a maximum height it can grow up to and then once reached, the user will have to scroll to see any additional tasks.

- Bug Fix: I noticed the navigation toolbar hover text was not formatted according to the light/dark mode switch. I changed it accordingly.

- Bug Fix: Made todo-list more responsive to size change. Admittedly, this needs a bit more work to look exactly like what we designed.

- Documentation Updates: Added specs for todo list, made some formatting changes (i.e. changed px to rem).


**What is the current behavior? (You can also link to an open issue here)**
Issues resolved/worked on include: #85 , #86 , #87 , #46 
Please see below images to understand changes. 

![Screen Shot 2024-12-08 at 5 17 23 PM](https://github.com/user-attachments/assets/270de723-0e53-4471-9d4b-8ff9c023c7f4)
![Screen Shot 2024-12-08 at 5 17 41 PM](https://github.com/user-attachments/assets/d478ef03-2d64-46c8-b83b-e2462c5fa4e8)
![Screen Shot 2024-12-08 at 5 17 58 PM](https://github.com/user-attachments/assets/1b517c13-4cd3-4f37-b503-ae28bd5ca5fd)
![Screen Shot 2024-12-08 at 5 18 09 PM](https://github.com/user-attachments/assets/029047a0-77d2-4e21-9312-5587c7db146e)
![Screen Shot 2024-12-08 at 5 18 20 PM](https://github.com/user-attachments/assets/84c96a23-ea4f-4f41-887c-6e935ce73ab7)

**What is the new behavior (if this is a feature change)?**
Incorporates the new feature of resizing for mobile friendliness - but is modified specifically for the todo-list. Also new feature of scrolling introduced to the todo list.

myrmanshelby commented on static/css/todo.css:
```diff
@@ -3,90 +3,103 @@
     background-color: var(--background-secondary-lm);
     font-family: var(--font-primary);
-    font-size: medium;
+    font-size: small;
```
Can you please use rem here to follow our best practices?

myrmanshelby commented on static/css/todo.css:
```diff
@@ -163,19 +178,33 @@
 
 /* Responsive design for smaller screens */
-@media (max-width: 768px) {
+@media (max-width: 48rem) {
```
max-width should stay in pixels, sorry should have clarified! rem is relative to the screen size and scales. Using px here allows the change to be consistent across screens

sbodmer01 commented:
Going to pull some changes from main also to stay up-to-date there. Don't merge quite yet, but please review and let me know if these changes are satisfactory!

myrmanshelby commented on templates/todoList.html:
```diff
@@ -4,10 +4,10 @@
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <title>To-Do List Component</title>
-  <link rel="stylesheet" href="{{ url_for('static', filename='css/todo.css') }}">
+  <link rel="stylesheet" href="{{ url_for('static', filename='css/todo.css') }}?v={{ time.time() }}">
```
what's the reason for the time code here?

sbodmer01 replied:
oops left that in on accident. That was just so that I could view changes in real time (added that at the very beginning but I don't need it anymore so I'll delete it) -- good catch!

myrmanshelby reviewed (changes requested):
Overall it looks really good, just requested a few changes! Could you also rebase with main before pushing, then we can review it again? It looks like there are merge conflicts, so we should handle the conflicts here rather than in the main branch

sbodmer01 commented:
Updated code and rebased with main -- it seems much more compatible with screen size changes! @myrmanshelby  or @ronit14235 , if any of you could review and see if these changes are good to go! 


![Screen Shot 2024-12-09 at 1 55 59 AM](https://github.com/user-attachments/assets/6cad8d62-98fe-45fc-8604-6bfc069c0211)
![Screen Shot 2024-12-09 at 1 56 09 AM](https://github.com/user-attachments/assets/b5d9d336-6baa-4fa5-acae-7d037b363e64)
![Screen Shot 2024-12-09 at 1 56 39 AM](https://github.com/user-attachments/assets/74d74319-ae89-4d1c-8a6c-6c3812c43e46)
![Screen Shot 2024-12-09 at 1 56 52 AM](https://github.com/user-attachments/assets/b5cec0e8-3a44-412c-a16b-103daa3c29c6)
![Screen Shot 2024-12-09 at 1 57 03 AM](https://github.com/user-attachments/assets/e9004d5c-033e-41cb-b192-c6e07dad54c7)

myrmanshelby commented on static/css/styles.css:
```diff
@@ -221,6 +221,59 @@ footer {
     cursor: pointer;
 }
 
+/* Tooltip Styling */
```
I removed this because we got rid of the tooltips, is there a reason we want to add the styling back in? Just want to minimize any unnecessary css code :)

sbodmer01 replied:
deleted! must have added it accidentally back through merging.

myrmanshelby commented on static/css/todo.css:
```diff
@@ -3,62 +3,76 @@
     box-sizing: border-box;
+
+    max-width: 100%; /*Ensures container doesn't exceed parent width*/
+    max-height: 16.25rem;
```
This is a pretty small max height for columns that could have a lot of entries. Do we want to increase this for a better UX?

sbodmer01 replied:
great idea! increased the max height to allow for visibility of 7 tasks -- any additional ones can be viewed by scrolling. Let me know if you think that's a good number of visible tasks or if we want them to see even more

myrmanshelby commented on static/css/todo.css:
```diff
@@ -3,62 +3,76 @@
+    overflow-y: hidden;
     box-sizing: border-box;
-    max-height: 100%; /* Ensure equal height */
+    max-height: 16.875rem; /* Ensure equal height */
```
same question as above, do we want the max height to be bigger? These boxes feel a little small to me but up to you!

myrmanshelby commented on static/css/todo.css:
```diff
@@ -147,7 +167,7 @@ input {
 /* Drag-and-drop feedback */
 .todo-item.drag-over {
-    border: 2px dashed var(--background-secondary-lm);
+    border: 125rem dashed var(--background-secondary-lm);
```
125rem is really large for a border, should this be 0.125rem? That's the rough conversion from px to rem

sbodmer01 replied:
oh, this portion is not used currently since drag and drop only appends the item to the end of the list currently. I'm currently working on that function, but it might be more like an enhancement.

myrmanshelby replied:
Got it, sounds good!

myrmanshelby reviewed (changes requested):
It's looking great, just a few nitpicky css things to change but nothing major. Also could you rebase again? It looks like there are still merge conflicts for some reason

sbodmer01 commented:
@myrmanshelby any chance you can review? Still working on additional functionality to allow for dragging and dropping to allow change in order of tasks -- current functionality allows drag and dropping but appends to the list. Core functionality here otherwise seems good to merge to main.

myrmanshelby reviewed (approved):
Looks fantastic, I think we're ready to merge!
